### PR TITLE
[Fluid Clean] remove nn.topk, nn.ctc_greedy_decoder, nn.im2sequence, nn.multiplex, nn.smooth_l1

### PR DESCRIPTION
**Pull request by heyanru01**
### PR types

Others
### PR changes

APIs
### Describe

remove nn.topk, nn.ctc_greedy_decoder, nn.im2sequence, nn.multiplex, nn.smooth_l1

### CODE REVIEW

**Review by csy0225 — COMMENTED** [review]
LGTM

File: python/paddle/fluid/layers/detection.py [hunk 1]
```diff
@@ -1836,7 +1836,8 @@ def __reshape_to_2d(var):
     location = __reshape_to_2d(location)
     target_bbox = __reshape_to_2d(target_bbox)
 
-    loc_loss = nn.smooth_l1(location, target_bbox)
+    smooth_l1_loss = paddle.nn.loss.SmoothL1Loss()
+    loc_loss = smooth_l1_loss(location, target_bbox)
```
Comment (jzhang533) [inline python/paddle/fluid/layers/detection.py:1692]: 为什么这里没用：`paddle.nn.functional.smooth_l1_loss`

File: python/paddle/fluid/tests/unittests/npu/test_smooth_l1_loss_op_npu.py [hunk 2]
```diff
@@ -138,22 +138,5 @@ def test_check_grad_ingore_y(self):
...
-            y1 = fluid.create_lod_tensor(
-                np.array([[-1]]), [[1]], fluid.NPUPlace(0)
-            )
-            self.assertRaises(TypeError, fluid.layers.smooth_l1, x1, y1)
-            # The input dtype of accuracy_op must be float32 or float64.
-            x2 = fluid.layers.data(name='x2', shape=[4], dtype="int32")
-            y2 = fluid.layers.data(name='x2', shape=[4], dtype="int32")
-            self.assertRaises(TypeError, fluid.layers.smooth_l1, x2, y2)
-
-
```
Comment (jzhang533) [inline python/paddle/fluid/tests/unittests/npu/test_smooth_l1_loss_op_npu.py:157]: 这个为什么删除了？

File: python/paddle/fluid/tests/unittests/test_ctc_align.py [hunk 3]
```diff
@@ -226,50 +225,6 @@ def config(self):
         )
 
 
-class TestCTCAlignOpApi(unittest.TestCase):
-    def test_api(self):
-        x = fluid.layers.data('x', shape=[4], dtype='float32')
-        y = fluid.layers.ctc_greedy_decoder(x, blank=0)
```
Comment (jzhang533) [inline python/paddle/fluid/tests/unittests/test_ctc_align.py:232]: 这个是确认paddle.nn下没有相应的新的API，计划直接删除了是吗？

File: python/paddle/fluid/tests/unittests/test_smooth_l1_loss_op.py [hunk 4]
```diff
@@ -124,22 +122,5 @@ def test_check_grad_ingore_y(self):
         )
 
 
-class TestSmoothL1LossOpError(unittest.TestCase):
```
Comment (jzhang533) [inline python/paddle/fluid/tests/unittests/test_smooth_l1_loss_op.py:127]: 同上

**Review by XiaoguangHu01 — DISMISSED** [review]
LGTM

File: python/paddle/fluid/layers/detection.py (hunk 1 above)
Comment (heyanru01) [reply python/paddle/fluid/layers/detection.py:1692]: 替换时发现paddle.nn.functional.smooth_l1_loss在检索目录下没有被使用过，同时test_smooth_l1_loss.py中相关部分早前已被替换成paddle.nn.loss.SmoothL1Loss的形式，此处与单测保持统一。

File: python/paddle/fluid/tests/unittests/test_ctc_align.py (hunk 3 above)
Comment (heyanru01) [reply python/paddle/fluid/tests/unittests/test_ctc_align.py:232]: 是的

File: python/paddle/fluid/tests/unittests/npu/test_smooth_l1_loss_op_npu.py (hunk 2 above)
Comment (heyanru01) [reply python/paddle/fluid/tests/unittests/npu/test_smooth_l1_loss_op_npu.py:157]: 应当是fluid.layers.smooth_l1(x, y, inside_weight=None, outside_weight=None, sigma=None)与paddle.nn.functional.smooth_l1_loss(input, label, reduction='mean', delta=1.0, name=None)的参数形式差异产生了报错，替换有问题，直接进行了删除。

File: python/paddle/fluid/tests/unittests/test_smooth_l1_loss_op.py (hunk 4 above)
Comment (heyanru01) [reply python/paddle/fluid/tests/unittests/test_smooth_l1_loss_op.py:127]: 应当是fluid.layers.smooth_l1(x, y, inside_weight=None, outside_weight=None, sigma=None)与paddle.nn.functional.smooth_l1_loss(input, label, reduction='mean', delta=1.0, name=None)的参数形式差异产生了报错，替换有问题，直接进行了删除。

**Review by jzhang533 — DISMISSED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by JiabinYang — APPROVED** [review]
LGTM

**Review by hong19860320 — APPROVED** [review]
LGTM
